packages/admin/cms-admin/src/generator/future/generateGrid.ts

```diff
@@ -57,6 +58,164 @@ function findInputObjectType(input: IntrospectionInputValue, schema: Introspecti
+}
+
+function getRequiredMutationArgs({ createMutationType }: { createMutationType: IntrospectionField }) {
+    const skipParams = ["input"];
+    const requiredParams: IntrospectionInputValue[] = [];
+    createMutationType.args.forEach((arg) => {
+        if (skipParams.includes(arg.name)) return;
+        if (arg.type.kind === "NON_NULL") {
+            requiredParams.push(arg);
+        }
+    });
+    return requiredParams;
+}
+
+function getRequiredGqlArgs({
+    config,
+    gridQueryType,
+    createMutationType,
+    gqlIntrospection,
+}: {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    config: GridConfig<any>;
+    gridQueryType: IntrospectionField;
+    createMutationType?: IntrospectionField;
+    gqlIntrospection: IntrospectionQuery;
+}) {
+    const params: IntrospectionInputValue[] = [];
+    getRequiredQueryArgs({ gridQueryType }).forEach((param) => {
+        params.push(param);
+    });
+    if (createMutationType) {
+        getRequiredMutationArgs({ createMutationType }).forEach((arg) => {
+            // skip similar params
+            if (params.find((addedArg) => addedArg.name === arg.name)) return;
+            params.push(arg);
+        });
+    }
+
+    const requiredParams: { type: string; name: string }[] = [];
+    params.forEach((arg) => {
+        if (arg.type.kind !== "NON_NULL") return;
+
+        if (arg.type.ofType.kind === "SCALAR") {
+            const nativeScalars = ["ID", "String", "Boolean", "Int", "Float", "DateTime", "JSONObject"];
+            if (!nativeScalars.includes(arg.type.ofType.name)) {
+                // probably just add to requiredParams-array, but needs to be tested
+                console.warn(
+                    `Currently not supported special SCALAR of type ${arg.type.ofType.name} in required param ${arg.name} of ${gridQueryType.name}${
+                        createMutationType ? ` or ${createMutationType.name}` : ``
+                    }`,
+                );
+            } else {
+                requiredParams.push({ type: arg.type.ofType.name, name: arg.name });
+            }
+        } else if (arg.type.ofType.kind === "INPUT_OBJECT") {
+            requiredParams.push({ type: arg.type.ofType.name, name: arg.name });
+        } else if (arg.type.ofType.kind === "LIST") {
+            console.warn(
+                `Currently not supported kind LIST in required param ${arg.name} of ${gridQueryType.name}${
+                    createMutationType ? ` or ${createMutationType.name}` : ``
+                }`,
+            );
+        } else if (arg.type.ofType.kind === "ENUM") {
+            console.warn(
+                `Currently not  supported kind ENUM in required param ${arg.name} of ${gridQueryType.name}${
+                    createMutationType ? ` or ${createMutationType.name}` : ``
+                }`,
+            );
+        } else {
+            throw new Error(`Not supported kind ${arg.type.ofType.kind}`);
+        }
```

Comment:
Why do we need this? Shouldn't the generator work with any GraphQL type?

Reply:
I would even say we reduce this in the first step to the bare minimum. The Api generator (which the admin generator is made for) currently does onl'y support `ID`s.

Reply:
I only removed LIST and ENUM because scalars other than ID and INPUT_OBJECT don't add any complexity and wont hurt.